src/main.ts

```diff
@@ -0,0 +1,34 @@
+import Vue from 'vue'
+import Dialog from './Dialog.vue'
+
+let pageLoadTime = Date.now()
+
+const isPasswordConfirmationRequired = (): boolean => {
+  const serverTimeDiff = pageLoadTime - (window.nc_pageLoad * 1000)
+  const timeSinceLogin = Date.now() - (serverTimeDiff + (window.nc_lastLogin * 1000))
+
+  // If timeSinceLogin > 30 minutes and user backend allows password confirmation
+  return (window.backendAllowsPasswordConfirmation && timeSinceLogin > 30 * 60 * 1000)
+}
+
+export const confirm = (): Promise<void> => {
+  const isDialogMounted = Boolean(document.getElementById('password-confirmation-dialog'))
+  if (isDialogMounted) {
+    return Promise.reject(new Error('Password confirmation dialog already mounted'))
+  }
+
+  if (!isPasswordConfirmationRequired()) {
+    return Promise.resolve()
+  }
+
+  const mountPoint = document.createElement('div')
+  mountPoint.setAttribute('id', 'password-confirmation-dialog')
+  document.querySelector('body')?.prepend(mountPoint)
+
+  new Vue({
+    el: '#password-confirmation-dialog',
+    render: h => h(Dialog),
+  })
+
+  return Promise.resolve()
```

Comment:
so this doesn't actually await the user interaction?

Reply:
I'd strongly advertise to keep the old behavior. Maybe you can listen to an event emitted from `Dialog` and only then resolve/reject the promise

Reply:
Background: password confirmation needs to be awaited before API requests can be sent that require the confirmed password